src/meson.build

```diff
@@ -0,0 +1,196 @@
+###### xdg-desktop-portal
+
+built_src_portal = gnome.gdbus_codegen('xdp-dbus',
+    sources: src_portal + src_flatpak_intf,
+    interface_prefix: 'org.freedesktop.portal',
+    namespace: 'XdpDbus',
+    docbook: 'portal',
+    autocleanup: 'none',
+)
+built_src_impl = gnome.gdbus_codegen('xdp-impl-dbus',
+    sources: src_portal_impl,
+    interface_prefix: 'org.freedesktop.impl.portal',
+    namespace: 'XdpDbusImpl',
+    docbook: 'portal',
+    autocleanup: 'none',
+)
+if have_geoclue
+    built_src_geoclue = gnome.gdbus_codegen('geoclue-dbus',
+        sources: 'org.freedesktop.GeoClue2.Client.xml',
+        interface_prefix: 'org.freedesktop.GeoClue2',
+        namespace: 'Geoclue',
+        autocleanup: 'none',
+    )
+else
+    built_src_geoclue = []
+endif
+
+built_resources = gnome.compile_resources('xdg-desktop-resources',
+    'xdg-desktop-portal.gresource.xml',
+    c_name: '_xdg_desktop',
+    source_dir: ['..'],
+)
+
+src_sd_escape = files('sd-escape.c', 'sd-escape.h')
+
+src_xdg_desktop_portal = [
+    'xdg-desktop-portal.c',
+    'file-chooser.c',
+    'file-chooser.h',
+    'glib-backports.c',
+    'glib-backports.h',
+    'open-uri.c',
+    'open-uri.h',
+    'print.c',
+    'print.h',
+    'memory-monitor.c',
+    'memory-monitor.h',
+    'network-monitor.c',
+    'network-monitor.h',
+    'power-profile-monitor.c',
+    'power-profile-monitor.h',
+    'proxy-resolver.c',
+    'proxy-resolver.h',
+    'screenshot.c',
+    'screenshot.h',
+    'notification.c',
+    'notification.h',
+    'inhibit.c',
+    'inhibit.h',
+    'device.c',
+    'device.h',
+    'account.c',
+    'account.h',
+    'realtime.c',
+    'realtime.h',
+    'request.c',
+    'request.h',
+    'call.c',
+    'call.h',
+    'documents.c',
+    'documents.h',
+    'permissions.c',
+    'permissions.h',
+    'email.c',
+    'email.h',
+    'settings.c',
+    'settings.h',
+    'session.c',
+    'session.h',
+    'trash.c',
+    'trash.h',
+    'wallpaper.c',
+    'wallpaper.h',
+    'xdp-utils.c',
+    'xdp-utils.h',
+    'background.c',
+    'background.h',
+    'gamemode.c',
+    'gamemode.h',
+    'secret.c',
+    'secret.h',
+    'flatpak-instance.c',
+    'flatpak-instance.h',
+    'portal-impl.h',
+    'portal-impl.c',
+    built_src_portal,
+    built_src_impl,
+    built_src_geoclue,
+    built_resources,
+]
```

Comment:
Do you really need to include header files here, and in the variables below? I don't think they're necessary, only the .c files

Reply:
https://mesonbuild.com/FAQ.html#do-i-need-to-add-my-headers-to-the-sources-list-like-in-autotools

You do not. If you do, it has no effect other than to assert at configure time that the files didn't mysteriously get deleted, and expose the headers identically to the `extra_files: ` kwarg, which is documented as:
> Not used for the build itself but are shown as source files in IDEs that group files by targets (such as Visual Studio)